pallets/cdd-offchain-worker/src/lib.rs

```diff
@@ -0,0 +1,246 @@
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as CddOffchainWorker {
+        /// Last block at which unsigned transaction get submitted with in the transaction pool
+        pub LastExtSubmittedAt get(last_extrinsic_submitted_at): T::BlockNumber;
+    }
+}
+
+decl_event! {
+    /// Events generated by the module.
+    pub enum Event<T>
+        where
+        BlockNumber = <T as frame_system::Trait>::BlockNumber,
+        AccountId = <T as frame_system::Trait>::AccountId,
+    {
+        /// Event generated when nominators get removed from the `Staking` storage
+        InvalidateNominators(BlockNumber, Vec<AccountId>),
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        /// initialize the default event for this module
+        fn deposit_event() = default;
+
+        /// Submit the list of invalidate nominators via unsigned transaction.
+        ///
+        /// we allow sending the transaction without a signature, and hence without paying any fees,
+        /// we need a way to make sure that only some transactions are accepted.
+        /// This function can be called only once every `T::CoolingInterval` blocks.
+        /// Transactions that call that function are de-duplicated on the pool level
+        /// via `validate_unsigned` implementation and also are rendered invalid if
+        /// the function has already been called in current "session".
+        ///
+        /// It's important to specify `weight` for unsigned calls as well, because even though
+        /// they don't charge fees, we still don't want a single block to contain unlimited
+        /// number of such transactions.
+        #[weight = SimpleDispatchInfo::FixedNormal(10_000_000)]
+        fn take_off_invalidate_nominators(origin, _block_number: T::BlockNumber, target: Vec<T::AccountId>, _signature: <T::SignerId as RuntimeAppPublic>::Signature) -> DispatchResult {
+            // This is an unsigned transaction so origin should be none
+            ensure_none(origin)?;
+            // apply a sanity check to know whether the length of target list is greater than 0 or not.
+            ensure!(target.len() > 0, Error::<T>::EmptyTargetList);
+            // call the mutable function that will change the list of nominators in the staking pallet
+            let _ = <pallet_staking::Module<T>>::unsafe_validate_cdd_expiry_nominators(target.clone())?;
+            // now change the block number at which this unsigned transaction get executed.
+            let current_block = <system::Module<T>>::block_number();
+            <LastExtSubmittedAt<T>>::put(current_block);
+            Self::deposit_event(RawEvent::InvalidateNominators(
+                <frame_system::Module<T>>::block_number(),
+                target,
+            ));
+            Ok(())
+        }
+
+
+        fn offchain_worker(block: T::BlockNumber) {
```

Comment:
Where is it configured how often this function should be called?  How is this function being automatically called?